## Building Permits dataset from data.calgary.ca

In [2]:
import pandas as pd

In [4]:
df = pd.read_csv('../data/raw/building_permits.csv')

In [5]:
# run all siz inspections commands
print('Shape:', df.shape)
print('\nFirst 5 rows:')
print(df.head())
print('\nColumn info:')
print(df.info())
print('\nNumeric summary:')
print(df.describe())


Shape: (484798, 30)

First 5 rows:
      PermitNum       StatusCurrent AppliedDate  IssuedDate CompletedDate  \
0  BP2013-09623           Completed  2013/07/08  2013/07/23    2014/01/31   
1  BP2024-19255           Completed  2024/09/11  2024/10/09    2025/04/16   
2  BP2025-05639  Pre Backfill Phase  2025/04/01  2025/04/16           NaN   
3  BP2023-10028           Completed  2023/06/26  2023/11/16    2024/02/22   
4  BP2023-11936           Completed  2023/07/25  2023/08/03    2024/02/27   

                          PermitType PermitTypeMapped  \
0         Single Construction Permit         Building   
1         Single Construction Permit         Building   
2         Single Construction Permit         Building   
3  Commercial / Multi Family Project         Building   
4         Single Construction Permit         Building   

                  PermitClass PermitClassGroup PermitClassMapped  ...  \
0  1106 - Single Family House    Single Family       Residential  ...   
1  1106 - Sin

In [6]:
# checking missing values
missing = df.isnull().sum()
missing_percent = (missing / len(df)) * 100
missing_report = pd.DataFrame({'missing_count': missing, 'missing_percent': missing_percent})
print(missing_report[missing_report['missing_count'] > 0])

                   missing_count  missing_percent
IssuedDate                 16192         3.339948
CompletedDate              19871         4.098821
PermitClass                   50         0.010314
WorkClassMapped             4288         0.884492
Description                43767         9.027884
ApplicantName             146950        30.311594
ContractorName            193711        39.957054
EstProjectCost             36635         7.556756
TotalSqFt                 332530        68.591455
OriginalAddress              906         0.186882
CommunityCode                906         0.186882
CommunityName                906         0.186882
LocationCount                906         0.186882
LocationTypes                906         0.186882
LocationAddresses            906         0.186882
LocationsWKT                 906         0.186882
LocationsGeoJSON             906         0.186882


## answer to the question in the DAY_002.md
1. 484798 rows and 30 columns
2. check cell above with the reports with 17 of 30 columns having missing values some over 60% some under 1%
3. one permit